unittests/backends/extra_payload_provider/horizon.yml

```diff
@@ -0,0 +1,29 @@
+# ============================================================================ #
+# Copyright (c) 2022 - 2025 NVIDIA Corporation & Affiliates.                   #
+# All rights reserved.                                                         #
+#                                                                              #
+# This source code and the accompanying materials are made available under     #
+# the terms of the Apache License 2.0 which accompanies this distribution.     #
+# ============================================================================ #
+
+name: horizon
+description: "CUDA-Q target for Horizon fake target."
+config:
+  # Tell DefaultQuantumPlatform what QPU subtype to use
+  platform-qpu: remote_rest
+  # Tell NVQ++ to generate glue code to set the target backend name
+  gen-target-backend: true
+  # Add preprocessor defines to compilation
+  preprocessor-defines: ["-D CUDAQ_QUANTUM_DEVICE"]
+  # Add the rest-qpu library to the link list
+  link-libs: ["-lcudaq-rest-qpu"]
+  # Define the lowering pipeline
+  platform-lowering-config: "classical-optimization-pipeline,globalize-array-values,func.func(canonicalize,state-prep),unitary-synthesis,func.func(canonicalize),apply-op-specialization{constant-prop=1},aggressive-early-inlining,expand-measurements,classical-optimization-pipeline,decomposition{enable-patterns=U3ToRotations},func.func(canonicalize,multicontrol-decomposition)"
+  # Tell the rest-qpu that we are generating QIR.
+  codegen-emission: qir-base
+  # Library mode is only for simulators, physical backends must turn this off
+  library-mode: false
+  # Specify the extra payload provider to use
+  extra-payload-provider: sunrise
```

Comment:
I was really hoping we would not need to manually modify the target definition to use extra payload providers. This could really lead to an explosion of unnecessary target definitions (they may be more than one valid extra payload provider per backend). If we want to keep this level of specificity, it may be better to think about the target-options, there at least a downstream library could add the required flags to nvq++ for the payload's unique feature.

Reply:
Good point. It makes a lot of sense to use target-options for this and I have updated the implementation.
Please let me know what you think.